src/main/resources/public/ts/controllers/administrator/order/confirm-generate-library.controller.ts

```diff
@@ -0,0 +1,73 @@
+import {Behaviours, ng, template, toasts} from 'entcore';
+import {IScope} from "angular";
+import {OrderRegion, OrdersRegion, Projects, Utils} from "../../../model";
+
+interface ViewModel {
+    extractSelectedOrders(): OrdersRegion;
+
+    generateLibraryOrder(): Promise<void>;
+
+    checkRejectedOrder(): boolean;
+}
+
+interface ICrreScope extends IScope {
+    vm: ViewModel;
+    display: any;
+    projects: Projects;
+}
+
+class Controller implements ng.IController, ViewModel {
+
+    constructor(private $scope: ICrreScope) {
+        this.$scope.vm = this;
+    }
+
+    $onInit() {
+    }
+
+    $onDestroy() {
+    }
+
+    extractSelectedOrders() : OrdersRegion {
+        if (this.$scope.display.allOrdersSelected || !this.$scope.projects.hasSelectedOrders()) {
+            return this.$scope.projects.extractAllOrders();
+        } else {
+            return this.$scope.projects.extractSelectedOrders();
+        }
+    }
+
+    generateLibraryOrder = async () : Promise<void> => {
+        const selectedOrders : OrdersRegion = this.extractSelectedOrders();
+        this.$scope.projects.all = [];
+        this.$scope.display.loading = true;
+        this.$scope.display.lightbox.waitingAdmin = false;
+        template.close('lightbox.waitingAdmin');
+        Utils.safeApply(this.$scope);
+        await selectedOrders.generateLibraryOrder().then(async data => {
+            if (data.status == 200) {
+                toasts.confirm('crre.order.region.library.create.message');
+                this.$scope.display.toggle = false;
+                Utils.safeApply(this.$scope);
+                this.$scope.display.allOrdersSelected = false;
+                Behaviours.applicationsBehaviours['crre'].SnipletScrollService
+                    .sendScroll(true);
+            } else {
+                if (data.status == 401){
+                    toasts.warning('crre.order.error.purse');
+                } else {
+                    toasts.warning('crre.order.region.library.create.err');
+                }
+                this.$scope.display.allOrdersSelected = false;
+                Behaviours.applicationsBehaviours['crre'].SnipletScrollService
+                    .sendScroll(true);
+            }
+        });
```

Comment:
Si ton services retourne une 4XX ou 5XX alors cela est considérer comme une erreur. Donc tu ne passe pas dans le then. Mais tu peux faire ce traitement dans la catch
```suggestion
        await selectedOrders.generateLibraryOrder().then(async data => {
            toasts.confirm('crre.order.region.library.create.message');
            this.$scope.display.toggle = false;
            Utils.safeApply(this.$scope);
            this.$scope.display.allOrdersSelected = false;
            Behaviours.applicationsBehaviours['crre'].SnipletScrollService
                .sendScroll(true);
        }).catch((error: AxiosError) => {
            if (error.response.status == 401){
                toasts.warning('crre.order.error.purse');
            } else {
                toasts.warning('crre.order.region.library.create.err');
            }
            this.$scope.display.allOrdersSelected = false;
            Behaviours.applicationsBehaviours['crre'].SnipletScrollService
                .sendScroll(true);
        });
```